WooCommerce/WooCommerceTests/Testing/XCTestCase+Wait.swift

```diff
@@ -21,4 +21,35 @@ extension XCTestCase {
         block(exp)
         wait(for: [exp], timeout: timeout)
     }
+
+    /// Creates an `XCTestExpectation` and waits until `condition` returns `true`.
+    ///
+    /// Example usage:
+    ///
+    /// ```
+    /// var valueThatIsUpdatedAsynchronously: Int = 0
+    ///
+    /// waitUntil {
+    ///     valueThatIsUpdatedAsynchronously > 5
+    /// }
+    /// ```
+    ///
+    func waitUntil(file: StaticString = #file,
+                   line: UInt = #line,
+                   timeout: TimeInterval = Constants.expectationTimeout,
+                   condition: @escaping (() -> Bool)) {
+        let predicate = NSPredicate { _, _ -> Bool in
+            return condition()
+        }
+
+        let exp = expectation(for: predicate, evaluatedWith: nil)
```

Comment:
TIL'ed [`expectation(for:evaluatedWith:handler:)`](https://developer.apple.com/documentation/xctest/xctestcase/1500569-expectation)!